### PR TITLE
Revolver ocelot

### PR DESCRIPTION
**Pull request by PopNotes**
:cl: PopNotes
balance: The .357 Revolver now does 45 damage per shot instead of the balls-nailed-to-the-wall 60.
balance:  .357 bullets are now faster to encourage firing at fleeing targets instead of the 'glorified melee weapon' meta.
balance: The .357 Revolver now only costs 10tc. What a steal!
balance: .357 speedloaders now only cost 2tc. Another steal!
balance: You can no longer print .357 speedloaders from the autolathe. What a drag!
soundadd: In light of weakening the Revolver, it now sounds way more powerful. Go figure.
/:cl:

Jesus fuck, this has been a long time coming.

### CODE REVIEW

**Comment by Pyko1** [comment]
about time revolver is one of the most strongest choice for pure murder power

**Comment by PopNotes** [comment]
Hol up, I can fix this

**Comment by banthebantz** [comment]
nah

**Comment by PopNotes** [comment]
that should fix it

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /obj/item/ammo_casing/a357
 	desc = "A .357 bullet casing."
 	caliber = "357"
-	projectile_type = /obj/item/projectile/bullet
+	projectile_type = /obj/item/projectile/magnum
```
Comment (tortellinitony) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: change this var in another file

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 2]
```diff
@@ -5,6 +5,7 @@
 	mag_type = /obj/item/ammo_box/magazine/internal/cylinder
 	origin_tech = "combat=3;materials=2"
 	casing_ejector = 0
+	fire_sound = 'hippiestation/sound/weapons/gunshot_magnum.ogg'
```
Comment (tortellinitony) [inline code/modules/projectiles/guns/ballistic/revolver.dm]: change this var in another file

File: code/modules/uplink/uplink_item.dm [hunk 3]
```diff
@@ -380,8 +380,8 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 
 /datum/uplink_item/ammo/pistol
 	name = "10mm Handgun Magazine"
-	desc = "An additional 8-round 10mm magazine; compatible with the Stechkin Pistol. These rounds \
-			are dirt cheap but are half as effective as .357 rounds."
+	desc = "An additional 8-round 10mm magazine; compatible with the Stechkin Pistol. These \
```
Comment (tortellinitony) [inline code/modules/uplink/uplink_item.dm]: another file

File: code/modules/projectiles/ammunition/ammo_casings.dm (hunk 1 above)
Comment (PopNotes) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: would making a new ammo_casings.dm and adding `projectile_type = /obj/item/projectile/magnum` work or would that just cause a conflict

Comment (tortellinitony) [reply code/modules/projectiles/ammunition/ammo_casings.dm]: yes have it look just like this except that's the only variable

**Comment by PopNotes** [comment]
That SHOULD be everything

**Comment by GuyonBroadway** [comment]
Add "Long silver bullets" to the uplink that have %100 chance to reflect twice.

**Comment by PopNotes** [comment]
ur pritty gud

**Comment by PopNotes** [comment]
@Carbonhell#4499 it's done my guy

File: hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/obj/item/weapon/gun/ballistic/revolver
+	fire_sound = 'hippiestation/sound/weapons/gunshot_magnum.ogg'
+
+/obj/item/weapon/gun/ballistic/revolver/attackby(obj/item/A, mob/user, params)
+	if(num_loaded)
```
Comment (tortellinitony) [inline hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: Add `..()`

Comment (PopNotes) [inline hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: where tho

Comment (PopNotes) [inline hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: w h e r e   t h o

Comment (ghost) [reply hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: first line

Comment (Carbonhell) [reply hippiestation/code/modules/projectiles/guns/ballistic/revolver.dm]: At the start of the proc

File: hippiestation/code/modules/uplink/uplink_item.dm [hunk 5]
```diff
@@ -119,3 +119,7 @@
 	cost = 5
 	item = /obj/item/weapon/bikehorn/golden/retardhorn
 	restricted_roles = list("Clown")
+
+/datum/uplink_item/ammo/pistol
+	desc = "An additional 8-round 10mm magazine; compatible with the Stechkin Pistol. These \
```
Comment (Carbonhell) [inline hippiestation/code/modules/uplink/uplink_item.dm:128]: ?????

Comment (PopNotes) [inline hippiestation/code/modules/uplink/uplink_item.dm]: ???????????????????????????

Comment (tortellinitony) [reply hippiestation/code/modules/uplink/uplink_item.dm:128]: he fixed the description because it isn't half as powerful anymore

Comment (GuyonBroadway) [reply hippiestation/code/modules/uplink/uplink_item.dm:128]: They never were IIRC, 10mm did 25 where the revolver did 60.

Comment (PopNotes) [inline hippiestation/code/modules/uplink/uplink_item.dm]: Wrong, 10mm casings use midbullet3, which does 30 damage

**Comment by McBawbaggings** [comment]
no
this nerfs the revolver's main function to be a good weapon since you're changing the crit in 2 to crit in 3.

the real issue is that you can print 18 bullets in cargo with 1 stack of 50 metal incredibly easily.
keep the revolver strong, make the ammo worth buying if you want to use it further but for the love of god stop the cargo traitor revolver meme.

i understand this affects cargo printing bullets but ultimately it's still gonna be absurdly strong regardless of the crit in 3 as opposed to crit in 2.

**Comment by McBawbaggings** [comment]
to add, even if cargo couldn't order in metal for dirt cheap, raiding engineering for their 2 stacks is essentially 24 bullets with low effort, that's 12 murders assuming perfect aim and crit in 2.

**Comment by banthebantz** [comment]
change bullets to be tablecraftable with chemicals and shit + framework for custom bullets ??????????????????????????????????

**Comment by McBawbaggings** [comment]
anything but "i print lots of free ammo xdddd i murder now...."
the fact you can buy ammo for the revolver but it's completely stupid unless you need it right now.

i understand the ammo exists in the autholathe for "detective" but don't you need to tamper with the gun and have it potentially misfire with those bullets or was that removed? but really the ammo in the autolathe exists for ez revolver ammo and it's fucking dumb

**Comment by McBawbaggings** [comment]
I will say though, the sound is really nice and suits it better than the current wimpy "pooft"

**Comment by PopNotes** [comment]
No concealable, spammable ranged weapon should be able to do **sixty damage per shot** with a capacity of six shots. End of discussion. That kind of firepower belongs only to huge weapons and melees.

Three shots to crit is still more than four, the stechkin's necessary shot amount- which is more than five, the glock's necessary shot amount. It's a natural progression. Even then, you're lucky to survive even two revolver shots, as you'll be at ten health with heavy bleeding. If you get/got punched twice, that's game over.

The insane damage is by far my biggest issue with the revolver, it's just too much. The ability to make infinite ammo from an autolathe is also a big problem, but contrary to your average powergamer's strawman of me I don't want to make the weapon completely worthless, so I'm keeping that in for now. If people decide it's STILL too powerful after this nerf, it's not a difficult change to remove ammo from the autolathe as well. Over-changing things is bad practice.

**Comment by GuyonBroadway** [comment]
Gotta agree with Mcbawb (on the sound and balance parts) the issue with revolverboning came from the fact you could just print more than enough ammo for it and by the time you ran out you had killed enough people to get your hands on an ID to go get more metal. 

The idea behind the revolver was that you could kill someone faster than anything else with the downside that you had limited ammo. If ammo was uncraftable it would be the weapon you use for one very quick murder. Not lots of very quick murders.

**Comment by McBawbaggings** [comment]
good change now i very like +1 ))))

**Comment by McBawbaggings** [comment]
image
bard approves

**Comment by zenek123** [comment]
Revert plz.

**Comment by PopNotes** [comment]
stfu

File: hippiestation/code/modules/projectiles/ammunition/ammo_casings.dm [hunk 6]
```diff
@@ -0,0 +1,2 @@
+/obj/item/ammo_casing/a357
+	projectile_type = /obj/item/projectile/magnum
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/projectiles/ammunition/ammo_casings.dm:2]: WOAH THERE SONNY JIM! THIS SHOULDA BEEN! obj/item/projectile/bullet/magnum

Comment (KayeArray) [reply hippiestation/code/modules/projectiles/ammunition/ammo_casings.dm:2]: THEY DIDN'T CHECK THIS BEFORE MERGING SCREECH

Comment (Carbonhell) [reply hippiestation/code/modules/projectiles/ammunition/ammo_casings.dm:2]: The fuck why didnt it runtime then

**Pull request merged**